rtree.go

```diff
@@ -30,15 +30,67 @@ type Rtree struct {
 	height      int
 }
 
-// NewTree creates a new R-tree instance.
-func NewTree(Dim, MinChildren, MaxChildren int) *Rtree {
-	rt := Rtree{Dim: Dim, MinChildren: MinChildren, MaxChildren: MaxChildren}
-	rt.height = 1
-	rt.root = &node{}
-	rt.root.entries = []entry{}
-	rt.root.leaf = true
-	rt.root.level = 1
-	return &rt
+// NewTree returns an Rtree. If the number of objects given on initialization
+// is larger than max, the Rtree will be initialized using the Overlap
+// Minimizing Top-down bulk-loading algorithm.
+func NewTree(dim, min, max int, objs ...Spatial) *Rtree {
+	rt := &Rtree{
+		Dim:         dim,
+		MinChildren: min,
+		MaxChildren: max,
+		height:      1,
+		root: &node{
+			entries: []entry{},
+			leaf:    true,
+			level:   1,
+		},
+	}
+
+	// if we have less objects than MaxChildren, we should just build the tree
+	// dynamically
+	n := len(objs)
+	if n <= rt.MaxChildren {
+		for _, obj := range objs {
+			rt.Insert(obj)
+		}
+
+		return rt
+	}
+
+	// create entries for all the objects
```

Comment:
could you extract all of this static initialization (the rest of this function) into a helper function?